### ISSUE TITLE
@babel/plugin-proposal-optional-chaining@7.0.0 compilation fails on expression that chains a function call with a chained argument

### ISSUE THREAD

**Issue by matthewadams** [babel/babel#9012]
## Bug Report

**Current Behavior**
Compilation fails on a legal optional chaining expression `const transpileBug = (a, b) => a?.bar(b?.foo)` with
```
ReferenceError: Container is falsy
    at NodePath._replaceWith (/Users/matthewadams/dev/node_modules/@babel/traverse/lib/path/replacement.js:188:11)
    at NodePath.replaceWith (/Users/matthewadams/dev/node_modules/@babel/traverse/lib/path/replacement.js:178:8)
    at optional (/Users/matthewadams/dev/node_modules/@babel/plugin-proposal-optional-chaining/lib/index.js:115:23)
    at PluginPass.OptionalCallExpression|OptionalMemberExpression (/Users/matthewadams/dev/node_modules/@babel/plugin-proposal-optional-chaining/lib/index.js:152:9)
    at newFn (/Users/matthewadams/dev/node_modules/@babel/traverse/lib/visitors.js:193:21)
    at NodePath._call (/Users/matthewadams/dev/node_modules/@babel/traverse/lib/path/context.js:53:20)
    at NodePath.call (/Users/matthewadams/dev/node_modules/@babel/traverse/lib/path/context.js:40:17)
    at NodePath.visit (/Users/matthewadams/dev/node_modules/@babel/traverse/lib/path/context.js:88:12)
    at TraversalContext.visitQueue (/Users/matthewadams/dev/node_modules/@babel/traverse/lib/context.js:118:16)
    at TraversalContext.visitMultiple (/Users/matthewadams/dev/node_modules/@babel/traverse/lib/context.js:85:17)
```

**Input Code**
```js
const transpileBug = (a, b) => a?.bar(b?.foo)

// the following is a successful workaround:
const foo = b?.foo
const transpileBug2 = (a, b) => a?.bar(foo)
```

**Expected behavior/code**
The given expression should not cause compiler to fail.

**Babel Configuration (.babelrc, package.json, cli command)**

```js
{
  "name": "foobar",
  "version": "0.1.0-dev.0",
  "engines": {
    "node": ">=10.12.0"
  },
  "scripts": {
    "build": "run-s test",
    "transpile": "run-s transpile-main transpile-test",
    "transpile-main": "babel --delete-dir-on-start --verbose --out-dir lib/main --copy-files src/main",
    "transpile-test": "babel --delete-dir-on-start --verbose --out-dir lib/test --copy-files src/test",
    "unit-integration": "nyc -x 'lib/test' mocha --opts mocha.opts 'lib/test/unit/**/*.spec.js' 'lib/test/integration/**/*.spec.js'",
    "integration": "nyc -x 'lib/test' mocha --opts mocha.opts 'lib/test/integration/**/*.spec.js'",
    "i": "mocha --opts mocha.opts 'src/test/integration/**/*.spec.js' | npx bunyan",
    "unit": "nyc -x 'lib/test' mocha --opts mocha.opts 'lib/test/unit/**/*.spec.js'",
    "u": "mocha --opts mocha.opts 'src/test/unit/**/*.spec.js' | npx bunyan",
    "test": "run-s transpile unit-integration lint",
    "lint": "standard --verbose 'src/**/*.js'",
    "format": "standard --fix 'src/**/*.js'"
  },
  "dependencies": {
    "@babel/polyfill": "7.0.0",
    "bluebird": "3.5.2",
    "bunyan": "1.8.12",
    "bunyaner": "1.0.1",
    "config": "2.0.1",
    "enumify": "1.0.4",
    "firebase-admin": "6.0.0",
    "flaverr": "1.9.2",
    "lodash": "4.17.11",
    "moment": "2.22.2",
    "moment-timezone": "0.5.21",
    "mutrait": "1.0.0-pre.9",
    "recurrify": "0.1.0-pre.1",
    "uuid": "3.3.2"
  },
  "devDependencies": {
    "@babel/cli": "7.1.2",
    "@babel/core": "7.1.2",
    "@babel/plugin-proposal-class-properties": "7.1.0",
    "@babel/plugin-proposal-decorators": "7.1.2",
    "@babel/plugin-proposal-optional-chaining": "7.0.0",
    "@babel/preset-env": "7.1.0",
    "@babel/register": "7.0.0",
    "babel-eslint": "10.0.1",
    "chai": "4.2.0",
    "config-cev-generator": "0.1.6",
    "dirty-chai": "2.0.1",
    "firebase-tools": "5.1.1",
    "fs-extra": "7.0.0",
    "intercept-stdout": "0.1.2",
    "mocha": "5.2.0",
    "npm-run-all": "4.1.3",
    "nyc": "13.1.0",
    "standard": "12.0.1"
  },
  "babel": {
    "sourceMaps": "inline",
    "retainLines": true,
    "presets": [
      [
        "@babel/preset-env",
        {
          "targets": {
            "node": true
          }
        }
      ]
    ],
    "plugins": [
      [
        "@babel/plugin-proposal-optional-chaining",
        {
          "loose": false
        }
      ],
      [
        "@babel/plugin-proposal-decorators",
        {
          "decoratorsBeforeExport": false
        }
      ],
      [
        "@babel/plugin-proposal-class-properties"
      ]
    ]
  },
  "standard": {
    "parser": "babel-eslint"
  }
}
```

**Environment**
- Babel version(s): 7.1.2 (@babel/core 7.1.2)
- Node/npm version: v10.12.0/6.4.1
- OS: macOS Mojave 10.14.1
- Monorepo: no
- How you are using Babel: `cli` by running `npm run transpile`

**Possible Solution**

**Additional context/Screenshots**

**Comment by harsh-tamr** [comment]
I've also encountered this issue

**Comment by jridgewell** [comment]
This was fixed by https://github.com/babel/babel/pull/9073, released in v7.2.0. [Repl builds fine](https://babeljs.io/repl/#?babili=false&browsers=Chrome%2072&build=&builtIns=false&spec=false&loose=false&code_lz=MYewdgzgLgBFBOBDSAHAlgGwKYCECuA5jALwwAUiANDAEYCUJAfDIgPwB0Ni8ZNHAZiBB0gA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=true&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=&prettier=true&targets=&version=7.4.5&externalPlugins=%40babel%2Fplugin-proposal-optional-chaining%407.2.0) with current version.
